libshaderc_spvc/src/spvcir_pass.cc

```diff
@@ -127,13 +194,241 @@ void SpvcIrPass::GenerateSpirvCrossIR(Instruction *inst) {
       break;
     }
 
+    // opcode: 16
+    case SpvOpExecutionMode: {
+      auto &execution = ir_->entry_points[inst->GetSingleWordInOperand(0u)];
+      auto mode =
+          static_cast<SpvExecutionMode>(inst->GetSingleWordInOperand(1u));
+      execution.flags.set(mode);
+
+      switch (mode) {
+        case SpvExecutionModeInvocations:
+          execution.invocations = inst->GetSingleWordInOperand(2u);
+          break;
+
+        case SpvExecutionModeLocalSize:
+          execution.workgroup_size.x = inst->GetSingleWordInOperand(2u);
+          execution.workgroup_size.y = inst->GetSingleWordInOperand(3u);
+          execution.workgroup_size.z = inst->GetSingleWordInOperand(4u);
+          break;
+
+        case SpvExecutionModeOutputVertices:
+          execution.output_vertices = inst->GetSingleWordInOperand(2u);
+          break;
+
+        default:
+          break;
```

Comment:
I would prefer this emits an error if there is at least one literal parameter, because that represents a case which is unhandled.  (E.g. this would occur if we ever anticipate using SPIR-V 1.4 and something like DenormPreserve)